rfc-113-expanding-draft-access-for-unauthenticated-users.md

```diff
@@ -0,0 +1,212 @@
+privacy.
+
+To provide access across a selection of pages an auth-bypass token would be
+stored in a session cookie this would allow maintaining this token across
+multiple requests without appending a query string (this cookie
+technique is already used for providing an [auth-bypass token to Asset
+Manager][asset-manager-auth-bypass]). To provide the auth-bypass id to
+supporting pages the Publishing API's link expansion system would be used
+for particular document and link types. As there isn't a linking system between
+assets and content it is impossible to maintain accurate links, thus an
+auth-bypass token can grant access to all draft assets for the duration of a
+session. Access limited assets must not be accessible in this circumstance
+unless the asset contains an auth-bypass id that matches the token.
+
+If there is a need to preview access limited content as part of a fact check
+then individual links will have to be provided for that content.
+
+### Consequences
+
+To support this functionality a number of changes are required in the GOV.UK
+stack, these are intended to be non-specific to the step-by-step problem and to
+support similar functionality for future GOV.UK content.
+
+- Authenticating Proxy will be changed to recognise auth-bypass tokens that are
+  present in cookies.
+- Asset Manager will be changed to recognise the presence of a flag in an
+  auth-bypass token that will grant an unauthenticated user access to a draft
+  asset that is not access limited. Fact checkers are not expected to be able
+  to exploit this access given sensitive assets should be access limited and
+  the URLs of assets are relatively unpredictable.
+- Currently Authenticating Proxy and Asset Manager do not check whether an
+  authenticated user has access to a resource if an auth-bypass token is
+  present. This introduces potential confusion when this token is provided by
+  a session cookie and could prevent an authenticated user viewing other parts
+  of draft GOV.UK. To counter this both apps will be changed to consider both
+  authentication and auth-bypass token when considering access to resources.
+- Content Store will be updated to allow an auth-bypass id to be present in the
+  links of a content item, this will be checked against a user provided
+  auth-bypass_id for content items that are not access limited.
+- Publishing API will be changed to support the sharing of auth-bypass ids for an
+  edition through link expansion rules.
+- Publishing API and Content Store will have their access limit code decoupled
+  from auth-bypass ids. This will be done to remove a frequent source of
+  confusion that conflates the separate processes of access limiting and
+  auth-bypass.
+- Collections Publisher will be changed to include a flag in auth-bypass tokens
+  that provide draft asset access. To reflect the increased access of a token
+  they will be given an expiry time (anticipated to be 1 month). Collections
```

Comment:
Where will that expiry happen? Is it when [creating the token](https://github.com/alphagov/authenticating-proxy/blob/56939ebb83481076811edcadff60f3a086d2961e/README.md#generating-a-auth-bypass-token)?

Reply:
Yup - my spike demonstrates that somewhat

https://github.com/alphagov/collections-publisher/blob/ebf9c7c4d4a9fe1378e9af61cb589c3f3cf6a281/app/lib/jwt_helper.rb#L8-L12